packages/authentication/test/acceptance/basic-auth.ts

```diff
@@ -91,8 +98,8 @@ describe('Basic Authentication', () => {
     });
   }
 
-  function givenAnApplication() {
-    app = new Application({
+  function givenAServer() {
+    server = new RestServer(server, {
```

Comment:
This is suspicious. Do you mean `server = new RestServer(app, {`?

Reply:
Yes, actually. Ty.

Reply:
In the grand scheme, the test no longer passes around an Application context, so it's not crucial, but circularly linking it serves no purpose, so I'll stub in an Application instance. :)

Reply:
What's the responsibility of `Server`? Here is my use case for discussion:

I have 3 APIs - a, b, c
- a: REST/HTTP
- b: REST/HTTPS for public ip & REST/HTTP for internal ip
- c: gRPC (HTTP2)

What terms will be involved to handle this situation?